Google.PowerShell/Sql/GcSqlInstanceCmdlets.cs

```diff
@@ -1388,4 +1388,159 @@ private Settings PopulateSetting(Settings newSettings)
+    public class FailoverGcSqlInstanceCmdlet : GcSqlCmdlet
+    {
+        private class ParameterSetNames
+        {
+            public const string ByName = "ByName";
+            public const string ByInstance = "ByInstance";
+        }
+
+        private class ErrorCodes
+        {
+            public const int InvalidSettingsVersion = 412;
+        }
+
+        private class ErrorMessages
+        {
+            public const string InvalidSettingsVersion =
+                "Input or retrieved settings version does not match current settings version for this instance.";
+        }
+
+        /// <summary>
+        /// <para type="description">
+        /// Name of the project in which the Instance resides.
+        /// Defaults to the Cloud SDK config for properties if not specified.
+        /// </para>
+        /// </summary>
+        [Parameter(ParameterSetName = ParameterSetNames.ByName)]
+        [ConfigPropertyName(CloudSdkSettings.CommonProperties.Project)]
+        public string Project { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The name/ID of the Instance resource to failover.
+        /// </para>
+        /// </summary>
+        [Parameter(ParameterSetName = ParameterSetNames.ByName, Mandatory = true, Position = 0)]
+        public string Instance { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The DatabaseInstance that describes the Instance we want to failover.
+        /// </para>
+        /// </summary>
+        [Parameter(ParameterSetName = ParameterSetNames.ByInstance, Mandatory = true, Position = 0,
+                   ValueFromPipeline = true)]
+        public DatabaseInstance InstanceObject { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The current settings version of the Instance. If not specified, it will be retrieved from the settings data
+        /// of the Instance. 
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        [ValidateNotNullOrEmpty]
+        public long? SettingsVersion { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            string projectName;
+            string instanceName;
+            switch (ParameterSetName)
+            {
+                case ParameterSetNames.ByName:
+                    projectName = Project;
+                    instanceName = Instance;
+                    break;
+                case ParameterSetNames.ByInstance:
+                    projectName = InstanceObject.Project;
+                    instanceName = InstanceObject.Name;
+                    break;
+                default:
+                    throw UnknownParameterSetException;
+            }
+
+            InstancesResource.GetRequest instanceGetRequest = Service.Instances.Get(projectName, instanceName);
+            DatabaseInstance instanceGetResponse = instanceGetRequest.Execute();
```

Comment:
This call is costly, and may not even be used. Consider not making this call unless you have to. Lazy<T> may be useful to you.